modin/core/execution/dask/common/engine_wrapper.py

```diff
@@ -133,3 +134,24 @@ def put(cls, data, **kwargs):
             data = UserDict(data)
         client = default_client()
         return client.scatter(data, **kwargs)
+
+    @classmethod
+    def wait(cls, obj_ids, num_returns=None):
+        """
+        Wait on the objects without materializing them (blocking operation).
+
+        Parameters
+        ----------
+        obj_ids : list, scalar
+        num_returns : int, optional
+        """
+        if not isinstance(obj_ids, list):
+            obj_ids = [obj_ids]
+        if num_returns is None:
+            num_returns = len(obj_ids)
+        if num_returns == len(obj_ids):
+            wait(obj_ids, return_when="ALL_COMPLETED")
+        else:
+            # Dask doesn't natively support `num_returns` as int
+            for _ in range(num_returns):
+                wait(obj_ids, return_when="FIRST_COMPLETED")
```

Comment:
We should return two lists as the original methods do because we may use the returned objects somewhere.

Take a look at the wait implementation in unidist for Dask backend. Maybe it is more beneficial. https://github.com/modin-project/unidist/blob/edbaadf80d5911d3723268950a42b776282448dc/unidist/core/backends/dask/backend.py#L101

Reply:
The feature you're talking about isn't used anywhere right now, which is why I didn't add it.
I suggest making this change when code appears that will use it.

Reply:
Agree with @anmyachev, let's add this functionality when we would really need it

Reply:
Ok, let's not add this functionality for now but please fix the wait implementation for Dask (look at the wait in unidist for Dask backend, for instance). It doesn't work correctly. Consider this example.

```python
from distributed import wait
from distributed import Client

c = Client()

def foo():
    import time
    time.sleep(10)
    return 1

done, not_done = wait([c.submit(foo) for _ in range(5)], return_when="FIRST_COMPLETED")
print(done)
{<Future: finished, type: int, key: foo2-c81de0e92e2c14ee29e3202c892d5f51>,
 <Future: finished, type: int, key: foo2-c81de0e92e2c14ee29e3202c892d5f51>,
 <Future: finished, type: int, key: foo2-c81de0e92e2c14ee29e3202c892d5f51>,
 <Future: finished, type: int, key: foo2-c81de0e92e2c14ee29e3202c892d5f51>,
 <Future: finished, type: int, key: foo2-c81de0e92e2c14ee29e3202c892d5f51>}
print(not_done)
set()
```

Reply:
@YarShev thanks for catching this! Ready for review.